src/id3/v2/tag.rs

```diff
@@ -547,21 +607,38 @@ impl SplitAndMergeTag for ID3v2Tag {
 		fn split_pair(
 			content: &str,
 			tag: &mut Tag,
-			current_key: ItemKey,
+			number_key: ItemKey,
 			total_key: ItemKey,
 		) -> Option<()> {
 			let mut split =
 				content.splitn(2, &[V4_MULTI_VALUE_SEPARATOR, NUMBER_PAIR_SEPARATOR][..]);
-			let current = split.next()?.to_string();
-			tag.items
-				.push(TagItem::new(current_key, ItemValue::Text(current)));
 
-			if let Some(total) = split.next() {
-				tag.items
-					.push(TagItem::new(total_key, ItemValue::Text(total.to_string())))
-			}
+			let number = split.next()?;
+			let total = split.next();
+
+			let is_valid = if let Some(total) = total {
+				content.len() == number.len() + 1 + total.len()
+					&& str::parse::<u32>(number).is_ok()
```

Comment:
By extracting a `parse_number()` function you could handle the whitespace trimming consistently. I consider "total" as a special kind of number, so the naming should be reasonable and we do not need to name it `parse_number_or_total()` .